### ISSUE TITLE
inconsistent error messages when including an extra space after "do"

### ISSUE THREAD

**Issue by trafnar** [imba/imba#600]
### Imba Version: x.x.x

these function calls all have a callback, and i included an extra space after `do` and before `(`

```imba
def fn callback do return true

# error: 'can't find name "bar"`
fn() do ({foo:bar})

# no error
fn() do ({foo})

# error: 'can't find name "foo"`
fn() do (foo)
```

It's hard to realize the mistake you've made in this situation, and the errors don't clarify it.

**Comment by somebee** [comment]
I guess this would be an argument for still using pipes for do arguments, since `do|a|  ...` and `do |a| ...` would always mean the same.

**Comment by andregoldstein** [comment]
> I guess this would be an argument for still using pipes for do arguments, since `do|a| ...` and `do |a| ...` would always mean the same.

I personally and purely aesthetically, much prefer the pipe, more ruby-esque I guess

**Comment by Bohooslav** [comment]
I liked pipes. Also, it would be easier to rewrite Scrimba code from Imba 1, where pipes was used, to imba 2. I think pipes should be as an option.

**Comment by somebee** [comment]
Do you like pipes as well @trafnar? I'm considering reverting back to good-ol pipes, and rather keep the `do(arg)` syntax around for compatibility. The main challenge with pipes was actually getting vscode to nicely auto-insert the matching pipe, but it should be possible to do.

**Comment by trafnar** [comment]
Yeah I'm fine with pipes for sure. Original issue here was just the inconsistent error reporting, not that it didn't work with the space.

**Comment by trafnar** [comment]
fwiw, I really prefer parens now, and always use them...
